yt/utilities/fortran_utils.py

```diff
@@ -69,15 +75,7 @@ def read_attrs(f, attrs, endian="="):
         if n == 1:
             v = v[0]
         if isinstance(a, tuple):
-            if len(a) != len(v):
-                raise OSError(
-                    "An error occurred while reading a Fortran "
-                    "record. Record length is not equal to expected "
-                    "length: %s %s",
-                    len(a),
-                    len(v),
-                )
-            for k, val in zip(a, v):
+            for k, val in zip(a, v, strict=True):
```

Comment:
not strictly backward compatible if we assume anyone relies on the exact exception type here, but I'm taking a bet that @cphyc doesn't mind.